src/compiler/types.ts

```diff
@@ -6123,8 +6123,15 @@ namespace ts {
         type: ESMap<string, number | string>;  // an object literal mapping named values to actual values
     }
 
+    /* @internal */
+    export interface AlternateModeDiagnostics {
+        diagnostic: DiagnosticMessage;
+        options: ReadonlySet<string>;
+    }
+
     /* @internal */
     export interface DidYouMeanOptionsDiagnostics {
+        alternateMode?: AlternateModeDiagnostics;

```

Comment:
Optional here but not in `ParseCommandLineWorkerDiagnostics`
Also these can just be two fields:
`alternateOptionsDeclarations` (pass from both compiler and build options (the opposite option declarations) you should need to create new set. 
`alternateOptionDiagnostic`

Reply:
@sheetalkamat I'm not following, sorry -- is there a reason the other modes should have this feature as well? I'd made this an optional object with required fields so to not change the type acquisition or watch mode options.

Reply:
You would need this when running tsc with build and option isnt supported with build?

Reply:
Agreed - I can add an _"{0} is only supported in build mode"_ error for those. I think for type acquisition and watch mode options it'd be cleaner to not require this?

Reply:
Done, thanks for the feedback @sheetalkamat!

I ended up making granular arrays for the lists of configurations so the CLI parser could use `.map(option => option.name)` in places instead of hardcoded lists.